search.js

```diff
@@ -0,0 +1,128 @@
+mainWrapperDiv = document.createElement("div");
+mainWrapperDiv.setAttribute("class", "mainWrapperDiv");
+searchDiv = document.createElement("div");
+searchDiv.setAttribute("class", "searchDiv");
+searchDiv.innerHTML = `<input type="text" class="searchBar" placeholder="Enter your search term here..."/>
+    <button class="searchButton"><i class="fa fa-search"></i> Search</button>`;
+mainWrapperDiv.appendChild(searchDiv);
+document.body.replaceChild(mainWrapperDiv, document.body.children[0]);
+contentWrapperDiv = document.createElement("div");
+contentWrapperDiv.setAttribute("class", "contentWrapperDiv");
+const searchBar = document.querySelector(".searchBar");
+const searchButton = document.querySelector(".searchButton");
+let result;
+const searchFunction = async () => {
+  let queryText = searchBar.value;
+  console.log(queryText);
+  await fetch(
+    "https://www.googleapis.com/youtube/v3/search?key=AIzaSyCk2ddyZVjqWUiLlsk5H4A3KXXLrLygsoc&type=video&part=snippet&maxResults=15&q=" +
+      queryText
+  )
+    .then((response) => response.json())
+    .then((data) => {
+      result = data;
+      displayCards();
+    })
+    .catch((err) => {
+      result = err;
+      displayError(err);
+    });
+};
+searchButton.addEventListener("click", searchFunction);
+
+let pageIndex = 0;
+let numberOfCards = 3;
+
+function displayCards(oldAnimation,newAnimation) {
+  numberOfCards = parseInt(document.querySelector(".mainWrapperDiv").offsetWidth/340);
+  console.log(numberOfCards);
+  let cards = document.querySelectorAll(".card");
+  if(cards[0]){
+    
+      [...cards].forEach((p) => {
+        p.parentNode.removeChild(p);})
+    }
+
+  
+  for(let i= pageIndex*numberOfCards;i<pageIndex*numberOfCards+numberOfCards&&i<result.items.length;i++){
```

Comment:
put the stopping condition outside of the loop. It would be evaluated everytime this way